### PR TITLE
Koen leijnse patch 1

### LINKED ISSUE #2: Migrate to qiskit 1.0
We are starting with the old code and it needs to be updated to work with Qiskit 1.0

### PR DESCRIPTION
**Pull request by koen-leijnse**
fixes #2 
I added a test file similar to the ones Carlos made in the Qnewtonraphson solver.
Some thing that can still be done: add different kind of observables to use with a sampler.

### CODE REVIEW

File: hhl_prototype/__init__.py [hunk 1]
```diff
@@ -72,21 +72,3 @@
 
 """
 
-from .hhl import HHL
```
Comment (NicoRenaud) [inline hhl_prototype/__init__.py:75]: you might want to keep:

```
 .solver.hhl import HHL
.solver.hhl_result import HHLResult
```

File: hhl_prototype/observables/absolute_average.py [hunk 2]
```diff
@@ -16,7 +16,9 @@
 import numpy as np
 
 from qiskit import QuantumCircuit
-from qiskit.opflow import I, Z, TensoredOp
+#from qiskit.opflow import I, Z, TensoredOp
```
Comment (NicoRenaud) [inline hhl_prototype/observables/absolute_average.py:19]: can you remove the commented lines ?

File: hhl_prototype/observables/linear_system_observable.py [hunk 3]
```diff
@@ -17,7 +17,8 @@
 import numpy as np
 
 from qiskit import QuantumCircuit
-from qiskit.opflow import TensoredOp
+#from qiskit.opflow import TensoredOp
```
Comment (NicoRenaud) [inline hhl_prototype/observables/linear_system_observable.py:20]: same here :)

File: hhl_prototype/observables/matrix_functional.py [hunk 4]
```diff
@@ -18,7 +18,8 @@
 
 from qiskit import QuantumCircuit
 from qiskit.quantum_info import Statevector
-from qiskit.opflow import I, Z, TensoredOp
+#from qiskit.opflow import I, Z, TensoredOp
```
Comment (NicoRenaud) [inline hhl_prototype/observables/matrix_functional.py:21]: commented lines

File: hhl_prototype/solver/hhl.py [hunk 5]
```diff
@@ -12,53 +12,24 @@
...
-
-    The HHL is a quantum algorithm to estimate a function of the solution with running time
-    complexity of :math:`\mathcal{ O }(\log(N)s^{2}\kappa^{2}/\epsilon)` when
-    :math:`A` is a Hermitian matrix under the assumptions of efficient oracles for loading the
-    data, Hamiltonian simulation and computing a function of the solution. This is an exponential
-    speed up in the size of the system, however one crucial remark to keep in mind is that the
-    classical algorithm returns the full solution, while the HHL can only approximate functions of
-    the solution vector.
+
+from qiskit.circuit.library import Isometry
```
Comment (NicoRenaud) [inline hhl_prototype/solver/hhl.py:22]: you import isometry twice i think

File: requirements.txt [hunk 6]
```diff
@@ -2,6 +2,7 @@ importlib_metadata
 mapomatic==0.6
 mthree==0.24
 pylatexenc
-qiskit-aer
-qiskit-ibmq-provider
-qiskit-terra>=0.21
+qiskit
+# qiskit-aer
```
Comment (NicoRenaud) [inline requirements.txt:6]: I think we  can remove the commented lines

**Review by NicoRenaud — APPROVED** [review]
Hey Koen,
nice PR ! I had a few comments but nothing serious so you can merge :)

**Comment by koen-leijnse** [comment]
I merged without working in your comments (can I just do it in the main branch?)

**Comment by Cmurilochem** [comment]
> I merged without working in your comments (can I just do it in the main branch?)

Hi @koen-leijnse. It happens and that is fine. You can do it. You could also create a new branch from the new master, make the changes, and then merge again yourself once you have done with the changes.

**Pull request merged**